src/test/java/co/rsk/bitcoinj/script/ScriptTest.java

```diff
@@ -472,84 +464,154 @@ public void numberBuilderNegative() {
+        Assert.assertEquals(expectedNumberOfRequiredSignatures, fastBridgeErpRedeemScript.getNumberOfSignaturesRequiredToSpend());
     }
 
     @Test
     public void getNumberOfSignaturesRequiredToSpend_no_fast_bridge_redeem_script() {
-        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(btcECKeyList);
-        Assert.assertEquals(2, redeemScript.getNumberOfSignaturesRequiredToSpend());
+        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(federationKeys);
+        Assert.assertEquals(expectedNumberOfRequiredSignatures, redeemScript.getNumberOfSignaturesRequiredToSpend());
     }
 
     @Test
     public void getSigInsertionIndex_fast_bridge_redeem_script() {
         byte[] data = Sha256Hash.of(new byte[]{1}).getBytes();
         Script fastBridgeRedeemScript = RedeemScriptUtils.createFastBridgeRedeemScript(
-            data, btcECKeyList);
+            data, federationKeys);
 
         testGetSigInsertionIndex(fastBridgeRedeemScript);
     }
 
     @Test
     public void getSigInsertionIndex_erp_redeem_script() {
         Script erpRedeemScript = RedeemScriptUtils.createErpRedeemScript(
-            btcECKeyList,
-            erpFedECKeyList,
+            federationKeys,
+            erpFederationKeys,
             500L
         );
 
         testGetSigInsertionIndex(erpRedeemScript);
     }
 
+    @Test
+    public void getSigInsertionIndexWhenP2shRedeemScriptShouldReturnIndex() {
+        Script redeemScript = RedeemScriptUtils.createP2shErpRedeemScript(
+            federationKeys,
+            erpFederationKeys,
+            600L
+        );
+
+        testGetSigInsertionIndex(redeemScript);
+    }
+
     @Test
     public void getSigInsertionIndex_fast_bridge_erp_redeem_script() {
         Script fastBridgeErpRedeemScript = RedeemScriptUtils.createFastBridgeErpRedeemScript(
-            btcECKeyList,
-            erpFedECKeyList,
+            federationKeys,
+            erpFederationKeys,
             500L,
             Sha256Hash.of(new byte[]{1}).getBytes()
         );
 
         testGetSigInsertionIndex(fastBridgeErpRedeemScript);
     }
 
+    @Test
+    public void getSigInsertionIndexWhenFlyoverP2shRedeemScriptShouldReturnIndex() {
+        Script flyoverP2shRedeemScript = RedeemScriptUtils.createFastBridgeP2shErpRedeemScript(
+            federationKeys,
+            erpFederationKeys,
+            500L,
+            Sha256Hash.of(new byte[]{1}).getBytes()
+        );
+
+        testGetSigInsertionIndex(flyoverP2shRedeemScript);
+    }
+
     @Test
     public void getSigInsertionIndex_no_fast_bridge_redeem_script() {
-        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(btcECKeyList);
+        Script redeemScript = RedeemScriptUtils.createStandardRedeemScript(federationKeys);
         testGetSigInsertionIndex(redeemScript);
     }
 
+    @Test
+    public void getSigInsertionIndexWhenEmptyScriptShouldThrowScriptException() {
```

Comment:
```suggestion
    public void getSigInsertionIndex_whenEmptyScript_shouldThrowScriptException() {
```

Reply:
Test name says one thing and implementation says another

Reply:
Updated.